### PR TITLE
[ISSUE-4029] Diff with only non null unread count.

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
### 🎯 Goal

Fix a bug where the unread count indicator does a little blinking. 

### 🛠 Implementation details

Now the diff doesn't accepts that the unread count is null, it should always have a value. 
### 🧪 Testing
This is very hard to reproduce, but sending a message in a channel that doesn't have unread count, could trigger the problem. 

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- ~[ ] New code is covered by unit tests~
- ~[ ] Comparison screenshots added for visual changes~
- ~[ ] Affected documentation updated (KDocs, docusaurus, tutorial)~

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF
giphy (1)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 ## stream-chat-android-client
 ### 🐞 Fixed
 - Rename of field for optional multi bundle push provider. Now projects with multiple push providers will correct correctly. [#4008](https://github.com/GetStream/stream-chat-android/pull/4008)
+- Fixed for of blinking unread count indicator. [#4030](https://github.com/GetStream/stream-chat-android/pull/4030)
```
Comment (MarinTolic) [inline CHANGELOG.md:18]:
```suggestion
- Fixed blinking unread count indicator. [#4030](https://github.com/GetStream/stream-chat-android/pull/4030)
```

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/extensions/internal/Channel.kt [hunk 2]
```diff
@@ -32,7 +32,7 @@ internal fun Channel.diff(other: Channel): ChannelListPayloadDiff {
         usersChanged = usersChanged,
         readStateChanged = read != other.read,
         lastMessageChanged = getLastMessage() != other.getLastMessage(),
-        unreadCountChanged = unreadCount != other.unreadCount,
+        unreadCountChanged = unreadCount != other.unreadCount && other.unreadCount != null,
```
Comment (kanat) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/extensions/internal/Channel.kt:35]: I wonder if `null` can be a valid value when there is no `unreadCount`

Comment (MarinTolic) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/extensions/internal/Channel.kt:35]: I tested a few scenarios including reading the channel from another device while the same user was logged in on two devices and saw no adverse effects.

**Review by MarinTolic — APPROVED** [review]
I could not reproduce the issue, but I tried to see if this PR caused any adverse effects on the unread count and it did not, so it might be a viable solution while apparently not causing any unwanted side-effects.

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works

**Comment by bychkovdmitry** [comment]
### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

**Review by kanat — APPROVED** [review]
LGTM

**Pull request merged**